quelpa.el

```diff
@@ -263,6 +263,42 @@ and return NEW-STAMP-INFO, otherwise return OLD-STAMP-INFO."
           new-stamp-info)
       old-stamp-info)))
 
+(defun quelpa-check-dir-hash (name config dir build-dir)
+  "Check if hash of DIR contents is different as stored in STAMP-FILE.
+If it is different save the new hash and timestamp to STAMP-FILE
+and return NEW-STAMP-INFO, otherwise return OLD-STAMP-INFO."
+  (let* ((hashes)
+         (new-content-hash)
+         (stamp-file (concat (expand-file-name
+                              (symbol-name name) build-dir) ".stamp"))
+         (time-stamp (replace-regexp-in-string
+                      ":" "" (replace-regexp-in-string
+                              ":0" "" (format-time-string "%Y%m%d.:%H%M%S"))))
+         (old-stamp-info (package-build--read-from-file stamp-file))
+         (new-stamp-info)
+         (old-content-hash (cdr old-stamp-info))
+         (old-time-stamp (car old-stamp-info))
+         (files (package-build--expand-source-file-list dir config)))
+
+    (setq files (mapcar (lambda (file) (expand-file-name file dir)) files))
+    (setq hashes (mapcar (lambda (file)
+                           (secure-hash
+                            'sha1 (concat file
+                                          (package-build--slurp-file file))))
+                         files))
+    (setq new-content-hash (secure-hash 'sha1 (mapconcat 'identity hashes "")))
+    (setq new-stamp-info (cons time-stamp new-content-hash))
+
+    (if (or (not old-content-hash)
+            (not (string= new-content-hash old-content-hash)))
+        (progn
```

Comment:
Won't this always evalute to `old-time-stamp` as long as `old-content-hash` exists (i.e. `old-content-hash` is non-nil)?